thinc/backends/cupy_ops.py

```diff
@@ -75,11 +75,23 @@ def asarray(self, data, dtype=None):
         dtype = {"dtype": dtype} if dtype is not None else {}
         if isinstance(data, cupy.ndarray):
             return self.xp.asarray(data, **dtype)
-        elif hasattr(data, "data_ptr"):
+        elif (
+            hasattr(data, "data_ptr")
+            and hasattr(data, "storage")
+            and hasattr(data, "device")
+        ):
             # Handles PyTorch Tensors
-            pointer = cupy.cuda.MemoryPointer(data.data_ptr())
-            shape = data.stride()
-            array = self.xp.ndarray(shape, memptr=pointer, **dtype)
+            assert data.device.type == "cuda", "Non-CUDA tensor passed to CupyOps"
```

Comment:
Since this method is part of the public API, it's probably better to throw a `ValueError` here?